mipengine/node/monetdb_interface/common_actions.py

```diff
@@ -195,7 +196,10 @@ def get_table_data(table_name: str) -> List[ColumnData]:
     data = list(zip(*data))
 
     columns_data = []
-    for current_column, current_values in zip(schema.columns, data):
+    # zip_longest is used because if there are no rows, no columns will be added as well
+    for current_column, current_values in itertools.zip_longest(
+        schema.columns, data, fillvalue=[]
```

Comment:
Why not fillvalue=None?

Reply:
I probably need to write this in a better way since you don't understand why this is as it is.

Currently, that we added the `check_rows` parameter, if it's false, a table could be returned without ANY rows.
This will return an empty list as the table data and when you will try to convert the table from row-stored to column-stored, it will end up badly.

Reply:
This doesn't work as you expect (probably) as it fills the shortest iterable with **refs to the same list**.

Reply:
@jassak The result of the `get_table_data` method is a `List[ColumnData]`. The empty lists are used in the Column

Reply:
> Am I missing something?

Yes, the fact that `ColumnData` is immutable doesn't mean the same for its attributes. In other words, if `ColumnData` has a list attribute, you cannot assign another list to it **but you can mutate the one it already has**. The reason is that a list is always a mutable data type.

> The data are only used for read purposes

Can you anticipate what can happen one year from now? Are you 100% sure no one will ever use it with some other purpose? Why not just fix the issue?